### PR TITLE
[flatbuffers] Update to 2.0.0

### LINKED ISSUE #17858: [flatbuffers] update to 2.0.0
Library name: flatbuffers

New version number: 2.0.0

Other information that may be useful (release notes, etc...)

> A summary of what is in this release since 1.12.0:
> 
>    - Support for optional scalars in most languages.
>    - Rust Verifier, Object API and FlexBuffers support, and many API improvements.
>    - Overhaul of TypeScript support.
>    - gRPC for many languages brought up to date.
>    - C++ compile time field reflection.
>    - JS FlexBuffers support.
>    - Kotlin FlexBuffers and platform support.
>    - Swift Object API and many Swift performance improvements.
>    - Lua supports LuaJIT and many more fixes.
>    - Many other fixes and smaller improvements, 371 commits since 1.12.0
> 
> Note, "2.0" does signify any kind of major overhaul of FlatBuffers, it is merely trying to be more semver compatible, and this release does have breaking changes for some languages much like all releases before it.

### PR DESCRIPTION
**Pull request by Wouterdek**
- #### What does your PR fix?  
  Fixes #17858

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  (it should same as before)

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

File: ports/flatbuffers/CONTROL [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Source: flatbuffers
-Version: 1.12.0
+Version: 2.0.0
 Port-Version: 2
```
Comment (JonLiu1993) [inline ports/flatbuffers/CONTROL:3]: Please delete "Port-Version"

File: versions/f-/flatbuffers.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "a8d0ffb0fa41c5c8bad5b955f71c4afa37a2db49",
+      "version-string": "2.0.0",
+      "port-version": 2
```
Comment (JonLiu1993) [inline versions/f-/flatbuffers.json:6]: Please set "port-version" to 0

**Review by JonLiu1993 — APPROVED** [review]

**Comment by Wouterdek** [comment]
Ok, I've changed the port version.

**Comment by PhoebeHui** [comment]
@Wouterdek, thanks for the PR!

The mnn failed on CI testing, could you please take a look? The affected triplets: x86-windows, x64-windows, x64-linux, x64-osx, x64-windows-static.


```
D:\buildtrees\mnn\src\1.1.0-59f7cd6d96.clean\schema\current\Type_generated.h(154): error C2440: 'initializing': cannot convert from 'const char *const [4]' to 'const int64_t *'
D:\buildtrees\mnn\src\1.1.0-59f7cd6d96.clean\schema\current\Type_generated.h(154): note: Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or function-style cast
D:\buildtrees\mnn\src\1.1.0-59f7cd6d96.clean\schema\current\Type_generated.h(212): error C2440: 'initializing': cannot convert from 'const char *const [22]' to 'const int64_t *'
D:\buildtrees\mnn\src\1.1.0-59f7cd6d96.clean\schema\current\Type_generated.h(212): note: Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or function-style cast
D:\buildtrees\mnn\src\1.1.0-59f7cd6d96.clean\schema\current\Tensor_generated.h(830): error C2440: 'initializing': cannot convert from 'const char *const [5]' to 'const int64_t *'
D:\buildtrees\mnn\src\1.1.0-59f7cd6d96.clean\schema\current\Tensor_generated.h(830): note: Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or function-style cast
D:\buildtrees\mnn\src\1.1.0-59f7cd6d96.clean\schema\current\Tensor_generated.h(863): error C2440: 'initializing': cannot convert from 'const char *const [9]' to 'const int64_t *'
D:\buildtrees\mnn\src\1.1.0-59f7cd6d96.clean\schema\current\Tensor_generated.h(863): note: Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or function-style cast
D:\buildtrees\mnn\src\1.1.0-59f7cd6d96.clean\schema\current\Tensor_generated.h(887): error C2440: 'initializing': cannot convert from 'const char *const [5]' to 'const int64_t *'
D:\buildtrees\mnn\src\1.1.0-59f7cd6d96.clean\schema\current\Tensor_generated.h(887): note: Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or function-style cast
D:\buildtrees\mnn\src\1.1.0-59f7cd6d96.clean\schema\current\Tensor_generated.h(922): error C2440: 'initializing': cannot convert from 'const char *const [9]' to 'const int64_t *'
D:\buildtrees\mnn\src\1.1.0-59f7cd6d96.clean\schema\current\Tensor_generated.h(922): note: Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or function-style cast
D:\buildtrees\mnn\src\1.1.0-59f7cd6d96.clean\schema\current\Tensor_generated.h(940): error C2440: 'initializing': cannot convert from 'const char *const [2]' to 'const int64_t *'
D:\buildtrees\mnn\src\1.1.0-59f7cd6d96.clean\schema\current\Tensor_generated.h(940): note: Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or function-style cast
```
[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/6512549/install-x64-windows-dbg-out.log)

**Comment by Wouterdek** [comment]
I've opened an issue [here](https://github.com/alibaba/MNN/issues/1476). Seems like there are some autogenerated files that need to be regenerated for the new flatbuffers version.

**Comment by JonLiu1993** [comment]
@Wouterdek ,Thanks for your PR!
It seems that the issue you submitted has not been assigned and resolved yet. Can you turn this PR into draft?

**Comment by JonLiu1993** [comment]
Ping @Wouterdek  for response, Is work still being done for this PR?

**Comment by Wouterdek** [comment]
> Ping @Wouterdek for response, Is work still being done for this PR?

No, I opened an issue with MNN, there is no movement there, and I ended up replacing VCPKG with git submodules in my project.

**Pull request closed without merging**